homeassistant/components/tuya/climate.py

```diff
@@ -436,6 +447,20 @@ def hvac_mode(self) -> HVACMode:
 
         return HVACMode.OFF
 
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return current hvac action."""
+        if self._attr_hvac_action:
```

Comment:
Why are we mixing up short hand with the property implementation here? 

We should use either, not both.

Reply:
Now, when you point me that, I'm start thinking that you are right :) and I should exclude messing with self._attr_hvac_action from here and from initialization

Reply:
This code is reworked